lib/cluster/s3_distribution_bucket.rb

```diff
@@ -74,7 +73,7 @@ def self.object_keys_to_delete(object)
       end
     end
 
-    def self.default_bucket_policy
+    def self.default_bucket_policy_for(bucket_name)
```

Comment:
Yes. How about if we have a S3BaseBucket class and S3DistributionBucket and S3ArchiveBucket subclasses where we specify the different policies?